### ISSUE TITLE
T59047: performance issue (regression?)

### ISSUE THREAD

**Issue by julienduroure** [KhronosGroup/glTF-Blender-IO#155]
https://developer.blender.org/T59047

Tests with default export options:
Tested on an old build I have on my laptop --> export in less than 1 minute
Tested with build of Dec 9th --> Still not finished after 15 minutes

We have to check if regression is on our side, or in blender api side

**Comment by julienduroure** [comment]
Tested on my old build (from Nov 30th), but with current version of addon code --> Still have the issue.

Problem seems to be on addon side, not on Blender API side

**Comment by julienduroure** [comment]
(Finished after 1h50!)

**Comment by emackey** [comment]
@julienduroure Did you get a successful export from the old build?  I tried an older copy of the addon, and it died with some error in less than a minute, but it didn't produce a valid glb file.

I suspect that the performance "regression" was caused by fixing the error, such that the whole file could export now, given enough time (hours!).

Still it seems that other formats can export much more quickly, so I'm hopeful that a performance bottleneck can be found.  I just don't think it can be found by looking at recent git history, instead it needs to be investigated with time-logging code that lets us find out which functions are costing the most time.
